packages/react-native-renderer/src/ReactNativeFiberRenderer.js

```diff
@@ -155,11 +173,28 @@ const NativeRenderer = ReactFiberReconciler({
     return false;
   },
 
-  getRootHostContext(): {} {
-    return emptyObject;
+  getRootHostContext(rootContainerInstance: Container): HostContext {
+    return __DEV__ ? {isInAParentText: false} : emptyObject;
   },
 
-  getChildHostContext(): {} {
+  getChildHostContext(
+    parentHostContext: HostContext,
+    type: string,
+  ): HostContext {
+    if (__DEV__) {
+      const oldIsInAParentText = parentHostContext.isInAParentText;
+      const newIsInAParentText =
+        type === 'RCTText' || type === 'RCTVirtualText';
```

Comment:
A new parent can never be a RCTVirtualText. Because of this:

https://github.com/facebook/react-native/blob/master/Libraries/Text/Text.js#L218-L222

Which highlights that this is in fact not just a DEV only contextual thing.

Reply:
> A new parent can never be a RCTVirtualText.

I'm not sure I understand what you're saying here.

`getChildHostContext()` can get called with `type === "RCTVirtualText"` in cases like:
```js
<Text>
  <Text>this is inside of a virtual text</Text>
</Text>
```

> Which highlights that this is in fact not just a DEV only contextual thing.

I don't understand. The "DEV only" bit here is the warning, not the runtime behavior.